DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -80,93 +125,124 @@ function Test-TargetResource
+    # Function will check the Address details are valid and do not conflict with
+    # Address family. Ensures interface exists.
+    # If any problems are detected an exception will be thrown.
+    [CmdLetBinding()]
     param
     (
-        [Parameter(Mandatory)]
-        [ValidateNotNullOrEmpty()]
-        [String[]]$Address,
+        [String]$Address,
 
         [Parameter(Mandatory)]
         [ValidateNotNullOrEmpty()]
         [String]$InterfaceAlias,
 
-        [Parameter(Mandatory)]
-        [ValidateSet("IPv4", "IPv6")]
-        [String]$AddressFamily,
-
-        [Switch]$Apply
+        [ValidateSet('IPv4', 'IPv6')]
+        [String]$AddressFamily = 'IPv4'
     )
-    $sa =$Address
-    $sa | Foreach-Object {
-             if(!([System.Net.Ipaddress]::TryParse($_, [ref]0)))
-             {
-                 throw "Server Address *$_* is not in the correct format. Please correct the Address in the configuration and try again"
-             }
-             if (([System.Net.IPAddress]$_).AddressFamily.ToString() -eq [System.Net.Sockets.AddressFamily]::InterNetwork.ToString())
-             {
-                if ($AddressFamily -ne "IPv4")
-                {
-                    throw "Server address *$_* is in IPv4 format, which does not match server address family $AddressFamily. Please correct either of them in the configuration and try again"
-                }
-             }
-             else
-             {
-                if ($AddressFamily -ne "IPv6")
-                {
-                    throw "Server address *$_* is in IPv6 format, which does not match server address family $AddressFamily. Please correct either of them in the configuration and try again"
-                }
-             }
-         }
-    try
-    {        
-        Write-Verbose -Message "Checking the DNS Server Address ..."
-        #Get the current IP Address based on the parameters given.
-        $currentAddress = (Get-DnsClientServerAddress -InterfaceAlias $InterfaceAlias -AddressFamily $AddressFamily -ErrorAction Stop).ServerAddresses
-
-        #Check if the Server addresses are the same as the desired addresses.
-        if(@(Compare-Object -ReferenceObject $currentAddress -DifferenceObject $Address -SyncWindow 0).Length -gt 0)
+    if(-not (Get-NetAdapter | Where-Object -Property Name -EQ $InterfaceAlias ))
+    {
+                throw ( @(
+                    "Interface $InterfaceAlias is not available. "
+                    'Please select a valid interface and try again'
+                    ) -join '' )
+    }
+    if ($Address)
+    {
+        if(-not ([System.Net.Ipaddress]::TryParse($Address, [ref]0)))
+        {
+                throw ( @(
+                    "Address $Address is not in the correct format. "
+                    'Please correct the Address parameter in the configuration and try again.'
+                    ) -join ''
+                )
+        }
+        if (([System.Net.IPAddress]$Address).AddressFamily.ToString() `
```

Comment:
Suggestion: you could change up this entire chunk to the following which would reduce some of the logic nesting that you currently have while maintaining the basic premise.

````powershell
function Test-ResourceProperty
{
    # Function will check the Address details are valid and do not conflict with
    # Address family. Ensures interface exists.
    # If any problems are detected an exception will be thrown.
    [CmdletBinding()]
    param
    (
        [String]$Address,

        [Parameter(Mandatory)]
        [ValidateNotNullOrEmpty()]
        [String]$InterfaceAlias,

        [ValidateSet('IPv4', 'IPv6')]
        [String]$AddressFamily = 'IPv4'
    )

    if( -not (Get-NetAdapter | Where-Object -Property Name -EQ $InterfaceAlias ))
    {
        throw ( @(
            "Interface $InterfaceAlias is not available. "
            'Please select a valid interface and try again'
            ) -join '' )
    }

    if( -not ([System.Net.Ipaddress]::TryParse($Address, [ref]0)))
    {
        throw ( @(
            "Address $Address is not in the correct format. "
            'Please correct the Address parameter in the configuration and try again.'
            ) -join ''
        )
    }

    if (([System.Net.IPAddress]$Address).AddressFamily.ToString() `
        -eq [System.Net.Sockets.AddressFamily]::InterNetwork.ToString() -and $AddressFamily -ne 'IPv4')
    {
        throw ( @(
                "Address $Address is in IPv4 format, which does not match "
                "server address family $AddressFamily. "
                'Please correct either of them in the configuration and try again.'
            ) -join '' )
    }

    if ($AddressFamily -ne 'IPv6')
    {
        throw ( @(
                "Address $Address is in IPv6 format, which does not match "
                "server address family $AddressFamily. "
                'Please correct either of them in the configuration and try again'
            ) -join '' )
    }
} # Test-ResourceProperty
````